master/server.go

```diff
@@ -16,11 +18,13 @@ type Job struct {
 func jobsHandler(w http.ResponseWriter, r *http.Request) {
 	var j Job
 	decoder := json.NewDecoder(r.Body)
-	decoder.Decode(&j)
+	if err := decoder.Decode(&j); err != nil {
+		return
```

Comment:
Better to print something at least, perhaps use the log package and print to stderr.